kerastuner/engine/oracle.py

```diff
@@ -333,11 +349,37 @@ def _get_oracle_fname(self):
             self._project_dir,
             'oracle.json')
 
-    def _compute_values_hash(self, values):
+    def _compute_values_hash(self, hps):
+        # Don't include conditional params that are not currently active.
+        values = {name: val for name, val in hps.values.items()
+                  if hps.is_active(name)}
         keys = sorted(values.keys())
         s = ''.join(str(k) + '=' + str(values[k]) for k in keys)
         return hashlib.sha256(s.encode('utf-8')).hexdigest()[:32]
 
+    def _random_populate_space(self, allow_collisions=False):
+        for _ in range(self._max_collisions):
```

Comment:
Perhaps this would read a bit more intuitively with

```
while collisions < self._max_collisions:
    ...
    if self._compute_values_hash(trial_hps) in self._tried_so_far:
        collisions += 1  # no need for comment since the meaning is transparent
```